clients/da-vinci-client/src/main/java/com/linkedin/davinci/blobtransfer/BlobSnapshotManager.java

```diff
@@ -163,13 +163,19 @@ protected Checkpoint createCheckpoint(RocksDB rocksDB) {
 
   /**
    * util method to create a snapshot for batch only
+   * It will check the snapshot directory and delete it if it exists, then generate a new snapshot
    */
   public static void createSnapshotForBatch(RocksDB rocksDB, String fullPathForPartitionDBSnapshot) {
     LOGGER.info("Creating snapshot for batch in directory: {}", fullPathForPartitionDBSnapshot);
 
-    if (fullPathForPartitionDBSnapshot == null || fullPathForPartitionDBSnapshot.isEmpty()) {
-      LOGGER.error("Snapshot directory {} is null or empty", fullPathForPartitionDBSnapshot);
-      return;
+    // clean up the snapshot directory if it exists
+    File partitionSnapshotDir = new File(fullPathForPartitionDBSnapshot);
+    if (partitionSnapshotDir.exists()) {
+      LOGGER.info("Snapshot directory already exists, deleting old snapshots at {}", fullPathForPartitionDBSnapshot);
+      if (!partitionSnapshotDir.delete()) {
```

Comment:
I'm a little confused on this. `File#delete` would only work for an empty directory. Do you want to delete the empty directory or the contents of it?

Reply:
good call! should not use File#delete, use FileUtils#deleteDirectory